recipes/mosquitto/2.x/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package C)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(mosquitto REQUIRED CONFIG)
+add_executable(${PROJECT_NAME} basic-1.c)
+target_link_libraries(${PROJECT_NAME} CONAN_PKG::mosquitto)
```

Comment:
You're using `CONAN_PKG::` of the `cmake` generator and using `cmake_find_package` with `find_package`.
The `find_package` is not needed.
Does mosquitto install cmake scripts? If so, then I think you should also set `self.cpp_info.names["cmake_find_package"]`.

Reply:
no, there is no cmake file, just pkgconfig files

Reply:
Then I would remove the `find_package` here and `cmake_find_package*` from the generators in `test_package/conanfile.py`.

Reply:
but this is what everyone (especially I) uses, cmake and cmake multi, so these should work 
and what is an alternative?

Reply:
The upstream project does not install any cmake scripts.
So you're relying on conan behavior that can change in the future if e.g. the upstream project uses other names for the files + targets..

This test does not use `mosquitto::mosquitto`, which it shouldn't anyways as upstream does not have it. So the `find_package` is not wanted here.